plugins/orchestrator/src/components/ExecuteWorkflowPage/ExecuteWorkflowPage.tsx

```diff
@@ -66,42 +76,61 @@ export const ExecuteWorkflowPage = (props: ExecuteWorkflowPageProps) => {
       });
   }, [orchestratorApi, workflowId]);
 
-  const onExecute = useCallback(async () => {
-    const parameters: Record<string, JsonValue> = {};
-    if (schemaResponse?.schema && formState) {
-      for (const key in formState) {
-        if (formState.hasOwnProperty(key)) {
-          const property = formState[key];
-          Object.assign(parameters, { [key]: property });
+  const onExecute = useCallback(
+    async (e: FormEvent) => {
+      e.preventDefault();
+
+      if (!schemaResponse) {
+        return;
+      }
+
+      let parameters: Record<string, JsonValue> = {};
+      if (schemaResponse.dataInputSchema?.mainSchema) {
+        if (!formRef.current?.validateForm()) {
+          return;
+        }
+        // Flatten parameters are only necessary for this case where we show one form for all ref schemas (steps).
+        parameters = flattenParametersFromFormState(formState);
```

Comment:
This comment isn't clear to me - why do we need this?

Reply:
Suppose you have the following input schema (the example is simplified):
```json
{
  "type": "object",
  "properties": {
    "ansible-job-template__ref-schema__GitHub_Repository_Info.json": {
      "$ref": "#/$defs/Provide information about the GitHub location_0",
      "type": "object"
    },
    "ansible-job-template__ref-schema__Ansible_Job_Definition.json": {
      "$ref": "#/$defs/Ansible Job Definition_1",
      "type": "object"
    }
  },
  "$defs": {
    "Ansible Job Definition_1": {
      "title": "Ansible Job Definition",
      "type": "object",
      "properties": {
        "name": {
          "title": "Name of the Ansible Job",
          "description": "A unique name for the Ansible Job",
          "type": "string"
        },
        "description": {
          "title": "Description",
          "description": "Provide a description of the Job to be launched",
          "type": "string"
        }
      },
      "required": [
        "name",
        "description"
      ]
    },
    "Provide information about the GitHub location_0": {
      "title": "Provide information about the GitHub location",
      "type": "object",
      "properties": {
        "githubOrg": {
          "title": "Organization Name",
          "description": "GitHub organization name",
          "type": "string"
        },
        "repoName": {
          "title": "Repository Name",
          "description": "GitHub repository name",
          "type": "string"
        }
      },
      "required": [
        "githubOrg",
        "repoName"
      ]
    }
  }
}
```

After filling up the form, the current implementation generates the following form data:
```json
{
  "ansible-job-template__ref-schema__GitHub_Repository_Info.json": {
    "githubOrg": "orgname",
    "repoName": "reponame"
  },
  "ansible-job-template__ref-schema__Ansible_Job_Definition.json": {
    "name": "jobname",
    "description": "desc"
  }
}
```

This is wrong. Properties must not be wrapped in their ref objects.
The correct form data to send to BE would be a flat object:
```json
{
  "githubOrg": "orgname",
  "repoName": "reponame",
  "name": "jobname",
  "description": "desc"
}
```

This is an issue that is happening in the current implementation given the nature of showing all steps at once.

The case where the input schema does not have refs are also covered, i.e. no need to flat anything, for instance:
```json
{
  "type": "object",
  "properties": {
    "language": {
      "title": "Language",
      "description": "Language to greet",
      "enum": [
        "English",
        "Spanish"
      ],
      "default": "English"
    },
    "name": {
      "title": "Name",
      "description": "Name of the person",
      "type": "string",
      "default": "John Doe"
    }
  },
  "required": [
    "name"
  ]
}
```

The corresponding form data for this one would be:
```json
{
  "language": "English",
  "name": "John Doe"
}
```

It probably won't be an issue for the multi-step form, but please keep it in mind.